.editorconfig

```diff
@@ -1,16 +1,79 @@
 root = true
 
-[*]
+
+## Repository Configurations
+############################
+[.{git*,editorconfig,*.yml}]
 charset = utf-8
 end_of_line = lf
 indent_style = space
+indent_size = unset
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[.gitmodules]
+indent_style = tab
+
+
+## Shell Scripts
+################
+[*.sh]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
```

Comment:
To the best of my knowledge, shell scripts should use tabs only for indentation, because spaces indents could lead to problems in some contexts — the matter is subject of dispute (as everything relating to the tabs-vs-space flamewars) but this seems to be the prevalent opinion, at least on long StackOverflow discussions.

But I'll set it to any value you suggest, the most important thing is to keep the code consistent.

Reply:
The reason why tabs are preferable is in case shell scripts are used to dynamically generate other scripts via pipes and redirections, where tabs are a safer option because common used tools will auto-strip leading tabs (but not spaces). 

An example:

https://www.howtobuildsoftware.com/index.php/how-do/ctWc/shell-sh-perforce-error-in-creating-p4-label-non-interactively-as-part-of-shell-script

Beside that, it boils down to a matter of personal choice (and the many flamewars on the subject didn't really bring any solution to the issue). I personally prefer using spaces in version controlled projects, for they create more readable diffs. But since in Make files tabs are mandatory, and shell scripts are sometimes used to manipulate Make files, I prefer to adopt tabs.

Reply:
Some interesting trivia on the subject:

- [Programmers who use spaces instead of tabs make more money](http://www.digitaljournal.com/tech-and-science/technology/programmers-who-use-spaces-instead-of-tabs-make-more-money/article/495295)

So, yes, I'm for spaces-indentation, but make an exception for shell scripts.

Reply:
Interesting, I didn't know. Thanks!